### PR TITLE
Migration of FHIR function to token-based auth

### LINKED ISSUE #5280: Use token-based auth for sending messages to Universal Pipeline

### PR DESCRIPTION
**Pull request by johanna-skylight**
# DEVOPS PULL REQUEST

## Related Issue

- Resolves #5280 

## Changes Proposed

- Terraform changes to create env variable `DATAHUB_FHIR_KEY` that maps to `datahub_fhir_key` in azure key vault for all environments. This new variable is the private key used to create the jwt token for the token-based auth flow with ReportStream.
- Terraform changes in the services folder to create env variable `REPORT_STREAM_BASE_URL` which maps to datahub_url in the key vault (already existing entry).
-  Implements method to create jwt token following the reportStream guidelines in v3.
- Implements method to obtain a bearer token from reportStream.
- Implements method to publish flu results to reportStream using token-based auth.
- Implements unit testing for all above.
- Removes the shared key implementation for the universal pipeline. Includes terraform changes to delete the variable `datahub_fhir_api_key`.

## Additional Information
This PR depends on 4 new secrets introduced in key vault.

## Testing
_(I have validated test events being successfully published in Dev4. Covid flows work as expected)_
- We can successfully publish flu results to reportStream universal pipeline using the token-based auth.
- We can successfully publish covid results as usual.

### CODE REVIEW

**Review by BobanL — COMMENTED** [review]
Overall, looks really good! Looking forward to seeing successful messages being sent!

**Review by emyl3 — COMMENTED** [review]
code-wise looks good to me! let me know if it's ready to test on a lower environment! Thank you for all your work on this!

File: ops/services/app_functions/report_stream_batched_publisher/functions/common/reportingHandlers.ts [hunk 1]
```diff
@@ -1,28 +1,56 @@
...
+  });
+
+  return fetch(`${REPORT_STREAM_BASE_URL}/api/waters`, {
+    method: "POST",
+    headers,
+    body: ndjsonTestEvents,
+  });
+}
+
+export async function reportToUniversalPipelineSharedKey(
```
Comment (mehansen) [inline ops/services/app_functions/report_stream_batched_publisher/functions/common/reportingHandlers.ts:46]: is the plan to clean this up eventually? if so let's make a cleanup ticket

**Review by mehansen — COMMENTED** [review]
looks great, thanks for your work on this. also looking forward to seeing how the testing goes on dev4

File: ops/services/app_functions/report_stream_batched_publisher/functions/common/reportingHandlers.ts (hunk 1 above)
Comment (johanna-skylight) [reply ops/services/app_functions/report_stream_batched_publisher/functions/common/reportingHandlers.ts]: I just cleaned up all the shared-key implementation

**Review by BobanL — APPROVED** [review]
All the changes look good to me and tested in Dev4!!

**Review by emyl3 — APPROVED** [review]
Tested on dev4 for multiplex and covid results and saw it was working in the logs! Nice work, Johanna!!!

**Review by rin-skylight — APPROVED** [review]
This looks good from a DevOps standpoint! Thank you for addressing the security alerts, as well! 😁

**Review by mehansen — APPROVED** [review]
🎉  looks good!

**Pull request merged**